optuna_dashboard/ts/components/DataGrid.tsx

```diff
@@ -177,15 +215,31 @@ function DataGrid<T>(props: {
           </TableBody>
         </Table>
       </TableContainer>
-      <TablePagination
-        rowsPerPageOptions={rowsPerPageOption}
-        component="div"
-        count={filteredRows.length}
-        rowsPerPage={rowsPerPage}
-        page={page}
-        onPageChange={handleChangePage}
-        onRowsPerPageChange={handleChangeRowsPerPage}
-      />
+      {filteredRows.length > 0 ? (
+        <>
+          {/* @ts-ignore */}
+          <Box display="flex" alignItems="center">
+            <TablePagination
+              rowsPerPageOptions={rowsPerPageOption}
+              component="div"
+              count={filteredRows.length}
+              rowsPerPage={rowsPerPage}
+              page={page}
+              labelDisplayedRows={({ page }) =>
+                `${page + 1} of ${maxPageNumber}`
+              }
+              onPageChange={handleChangePage}
+              onRowsPerPageChange={handleChangeRowsPerPage}
+            />
+            {maxPageNumber > 4 ? (
```

Comment:
I will quickly leave a comment before I forget.
For the other comments, I will address then when I'm back to the work!
The reason is very simple: When we move from page X to Y by the text field, we need three actions. (1) Click the text field, (2) Type a number, (3) Hit the enter key.
Meanwhile when we would like to move any pages for 4-page data grid, we need to make less than or equal to three actions. Namely, we need to click the prev/next key less than or equal to three times.
That's why it's unnecessary (or at least, not more efficient ) to use the text field in this case.

Reply:
Is it better to let it stay always?
Plus, is there any other stuff to address?

Reply:
I understand it. But the reason is not so obvious if we do not put code comment there. We can choose the following options:

1. Add comments for 4,
2. Change the number from 4 to 2 (because 2 is much easier to understand the reason), or
3. Always show it.